src/_igraph/graphobject.c

```diff
@@ -5017,80 +5016,65 @@ PyObject *igraphmodule_Graph_get_shortest_paths(igraphmodule_GraphObject *
     return NULL;
   }
 
+  /* The starting point is a single vertex, but the end is a list
+   * of vertices, so we need one shortest path per target vertex */
   if (igraph_vs_size(&self->g, &to, &no_of_target_nodes)) {
     if (weights) { igraph_vector_destroy(weights); free(weights); }
     igraph_vs_destroy(&to);
     igraphmodule_handle_igraph_error();
     return NULL;
   }
 
-  ptrvec = (igraph_vector_ptr_t *) calloc(1, sizeof(igraph_vector_ptr_t));
-  if (!ptrvec) {
+  /* Memory for the pointer */
+  veclist = (igraph_vector_int_list_t *) calloc(1, sizeof(igraph_vector_int_list_t));
+  if (!veclist) {
     PyErr_SetString(PyExc_MemoryError, "");
     if (weights) { igraph_vector_destroy(weights); free(weights); }
     igraph_vs_destroy(&to);
     return NULL;
   }
 
-  if (igraph_vector_ptr_init(ptrvec, no_of_target_nodes)) {
+  /* Initialize the vector_int_list itself, size is managed internally
+   * by the C core function */
+  if (igraph_vector_int_list_init(veclist, 0)) {
     PyErr_SetString(PyExc_MemoryError, "");
-    free(ptrvec);
+    free(veclist);
     if (weights) { igraph_vector_destroy(weights); free(weights); }
     igraph_vs_destroy(&to);
     return NULL;
   }
 
-  res = (igraph_vector_int_t *) calloc(no_of_target_nodes, sizeof(igraph_vector_int_t));
-  if (!res) {
-    PyErr_SetString(PyExc_MemoryError, "");
-    igraph_vector_ptr_destroy(ptrvec); free(ptrvec);
-    if (weights) { igraph_vector_destroy(weights); free(weights); }
-    igraph_vs_destroy(&to);
-    return NULL;
-  }
-
-  for (i = 0; i < no_of_target_nodes; i++) {
-    VECTOR(*ptrvec)[i] = &res[i];
-    igraph_vector_int_init(&res[i], 0);
-  }
-
-  if (igraph_get_shortest_paths_dijkstra(&self->g, use_edges ? 0 : ptrvec,
-        use_edges ? ptrvec : 0, from, to, weights, mode, 0, 0)) {
+  /* Call the C function */
+  if (igraph_get_shortest_paths_dijkstra(&self->g, use_edges ? 0 : veclist,
+        use_edges ? veclist : 0, from, to, weights, mode, 0, 0)) {
     igraphmodule_handle_igraph_error();
-    for (j = 0; j < no_of_target_nodes; j++) igraph_vector_int_destroy(&res[j]);
-    free(res);
-    igraph_vector_ptr_destroy(ptrvec); free(ptrvec);
+    igraph_vector_int_list_destroy(veclist); free(veclist);
     if (weights) { igraph_vector_destroy(weights); free(weights); }
     igraph_vs_destroy(&to);
     return NULL;
   }
 
-  igraph_vector_ptr_destroy(ptrvec); free(ptrvec);
-  if (weights) { igraph_vector_destroy(weights); free(weights); }
+  /* We don't need this anymore, the result is in veclist */
   igraph_vs_destroy(&to);
 
+  /* Make empty Python list of paths */
   list = PyList_New(no_of_target_nodes);
   if (!list) {
-    for (j = 0; j < no_of_target_nodes; j++) igraph_vector_int_destroy(&res[j]);
-    free(res);
     return NULL;
   }
 
+  /* Fill list of paths */
```

Comment:
We have an `igraph_vector_int_list_to_PyObject()` function in `convert.c`; can't we use that here?